activerecord/lib/active_record/connection_adapters/abstract/schema_statements.rb

```diff
@@ -985,13 +985,34 @@ def foreign_key_options(from_table, to_table, options) # :nodoc:
       end
 
       def dump_schema_information #:nodoc:
+        versions = ActiveRecord::SchemaMigration.order('version').pluck(:version)
+        insert_versions(versions)
+      end
+
+      def insert_versions(versions)
+        if supports_multi_insert?
+          multi_value_version_insert(versions)
+        else
+          single_value_version_insert(versions)
+        end
```

Comment:
Still don't see separating these methods as buying us anything. It expands our API footprint and suggests that we actually use them externally.